docs/docs/guides/web3_config/index.md

```diff
@@ -201,6 +202,610 @@ console.log(web3.getContextObject().config)
+The `transactionSendTimeout` is used to wait for Ethereum Node to return the sent transaction result. Note: If the RPC call stuck at the Node and therefor timed-out, the transaction may still be pending or even mined by the Network. We recommend checking the pending transactions in such a case.
+
+The default value of `transactionSendTimeout` is 750 seconds (12.5 minutes).
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.transactionSendTimeout = 750000; // 750000 ms = 750 s
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `transactionSendTimeout` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.transactionSendTimeout  = 750000; // 750000 ms = 750 s
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+### transactionConfirmationPollingInterval
+The `transactionConfirmationPollingInterval` is deprecated. Please use `transactionReceiptPollingInterval` or `transactionPollingInterval` instead.
+
+### blockHeaderTimeout
+The `blockHeaderTimeout` is used over socket-based connections. After sending a transaction, it will listen for the appearance of new blocks and proceed with subsequent operations based on the transaction results within them. This option defines the amount seconds it should wait for 'newBlockHeaders' event before falling back to polling to fetch transaction receipt.
+
+The default value of `blockHeaderTimeout` is 10 seconds.
+
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.blockHeaderTimeout = 10; // 10 s
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `blockHeaderTimeout` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.blockHeaderTimeout = 10; // 10 s
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+### maxListenersWarningThreshold
+The `maxListenersWarningThreshold` is used to set the `maxListeners` property in `EventEmitter`.
+
+The default value of `maxListenersWarningThreshold` is 100.
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.maxListenersWarningThreshold = 100;
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `maxListenersWarningThreshold` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.maxListenersWarningThreshold = 100;
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+### contractDataInputFill
+The `contractDataInputFill` will allow you to set the hash of the method signature and encoded parameters to the property either `data`, `input` or `both` within your contract. This will affect the contracts send, call and estimateGas methods.
```

Comment:
It would be nice to link to the API docs for the methods as well.

```suggestion
The `contractDataInputFill` will allow you to set the hash of the method signature and encoded parameters to the property either `data`, `input` or `both` within your contract. This will affect the contracts `send`, `call` and `estimateGas` methods.
```

Reply:
I tried to link these as well, however, it seems there is no separate link for these methods. https://docs.web3js.org/api/web3-eth-contract/class/Contract

Reply:
I find the link [here](https://docs.web3js.org/libdocs/Contract#send) 😄